homeassistant/components/sharkiq/vacuum.py

```diff
@@ -66,12 +66,27 @@
 ATTR_RSSI = "rssi"
 
 
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Shark IQ vacuum cleaner."""
+    coordinator: SharkIqUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    devices: Iterable["SharkIqVacuum"] = coordinator.shark_vacs.values()
+    device_names = [d.name for d in devices]
+    LOGGER.debug(
+        "Found %d Shark IQ device(s): %s",
+        len(device_names),
+        ", ".join([d.name for d in devices]),
+    )
+    async_add_entities([SharkVacuumEntity(d, coordinator) for d in devices])
+
+
 class SharkVacuumEntity(CoordinatorEntity, StateVacuumEntity):
     """Shark IQ vacuum entity."""
 
     def __init__(self, sharkiq: SharkIqVacuum, coordinator: SharkIqUpdateCoordinator):
         """Create a new SharkVacuumEntity."""
         super().__init__(coordinator)
+        # This should never be a problem, but checking doesn't hurt
```

Comment:
Let's remove this check. We're creating the `coordinator.shark_vacs` dict ourselves so we don't need this check.